src/Chapter3.hs

```diff
@@ -480,6 +509,50 @@ After defining the city, implement the following functions:
    and at least 10 living __people__ inside in all houses of the city in total.
 -}
 
+data Castle
+  = None
+  | Castle {castleName :: String}
+  | CastleWithWalls {castleName :: String}
+  deriving (Show)
+
+-- Why can't we use 1 | 2 | 3 | 4 here? 
+data HousePeople
+  = One
+  | Two
+  | Three
+  | Four
+  deriving (Show)
```

Comment:
Constructors in Haskell must start with an uppercase letter. Constants `1` and `2` and just numeric literals, and you can't create a new type in Haskell but reusing values of some other types with `|` as you wrote.